packages/clerk-js/src/ui/signIn/SignInFactorOne.tsx

```diff
@@ -1,23 +1,24 @@
 import { SignInFactor } from '@clerk/types';
 import React from 'react';
-import { LoadingScreen, shouldDisableStrategy, withRedirectToHome } from 'ui/common';
-import { useCoreSignIn, useEnvironment, useSignInContext } from 'ui/contexts';
+import { LoadingScreen, withRedirectToHome } from 'ui/common';
+import { useCoreSignIn, useEnvironment } from 'ui/contexts';
 import { AllFirstFactorStrategies, SignInFactorOneInputBased, SignInFactorOneMagicLink } from 'ui/signIn/factorOne';
 
 import { ErrorScreen } from './strategies';
-import { determineStartingSignInFactor } from './utils';
+import { determineStartingSignInFactor, isContactInfoRelatedStrategy } from './utils';
 
 function _SignInFactorOne(): JSX.Element {
-  const signInContext = useSignInContext();
   const signIn = useCoreSignIn();
   const { displayConfig } = useEnvironment();
   const { preferredSignInStrategy } = displayConfig;
-  const availableFactors = signIn.supportedFirstFactors.filter(f => !shouldDisableStrategy(signInContext, f.strategy));
+  const availableFactors = signIn.supportedFirstFactors;
   const [lastUsedFactor, setLastUsedFactor] = React.useState<SignInFactor | null>(null);
   const [currentFactor, setCurrentFactor] = React.useState<SignInFactor | undefined | null>(() => {
     return determineStartingSignInFactor(availableFactors, signIn.identifier, preferredSignInStrategy);
   });
-  const [showAllStrategies, setShowAllStrategies] = React.useState<boolean>(!currentFactor);
+  const [showAllStrategies, setShowAllStrategies] = React.useState<boolean>(
+    !currentFactor || !isContactInfoRelatedStrategy(currentFactor.strategy),
```

Comment:
Not related to this PR, but in the future we should discuss about refactoring all this logic into a Factor (or similar) object. Some components feel like they have too much logic, lets discuss after components v2 🙃